core/src/main/java/io/questdb/griffin/SqlResourceLimiterImpl.java

```diff
@@ -0,0 +1,24 @@
+package io.questdb.griffin;
+
+import io.questdb.griffin.engine.LimitOverflowException;
+
+public class SqlResourceLimiterImpl implements SqlResourceLimiter {
+    private long maxInMemoryRows;
+    private SqlExecutionInterruptor interruptor;
+
+    @Override
+    public void checkLimits(long nRows) {
+        if (nRows < maxInMemoryRows) {
+            interruptor.checkInterrupted();
```

Comment:
I see the reason for checks on every map operation. I guess idea doing this on map was to do it in one place? I have a disappointment here, we have other operations needing interruptions that do not involve map at all. I would suggest to decouple limit check and interrupt.